### PR TITLE
chore: templates

### PR DESCRIPTION
**Pull request by StrangeGirlMurph**
added templates for the commands and events and fixed the type error with anttis suggestion

### CODE REVIEW

File: src/events/EventSample.disable [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// set the function name to the event name 
+// and edit the parameter to fit in its type and nameto the event
+export default async function interactionName(agrs: unknown) {
+    // do things if the event occures 
+}
```
Comment (Chicken) [inline src/events/EventSample.disable:5]:
```suggestion
// and edit the parameter to fit in its type and name to the event
export default async function interactionName(args: unknown) {
    // do things if the event occurs 
}

```

File: src/events/ready.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ async function updateRegisteredCommands(client: Client) {
 	}
 }
 
-function commandsEqual(c1: ApplicationCommand, c2: SlashCommandBuilder): boolean {
+function commandsEqual(c1: ApplicationCommand, c2: SlashCommandBuilder | Omit<SlashCommandBuilder, "addSubcommandGroup" | "addSubcommand">): boolean {
```
Comment (Chicken) [inline src/events/ready.ts]: Probably too long, run prettier

File: src/commands/CommandSample.disable [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { Command } from "../Command";
+import { CommandInteraction } from "discord.js";
+import { SlashCommandBuilder } from "@discordjs/builders";
+
+// set the class name, the export at the bottom and the file name to your desired command name
+class CommandName extends Command {
+    constructor() {
+        super("bunny"); // the name under which the bot internally stores your command
```
Comment (Chicken) [inline src/commands/CommandSample.disable:8]: Actually this should be the same as the named set in `register`

File: src/commands/CommandSample.disable [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    async execute(interaction: CommandInteraction): Promise<void> {
+        // put the logic of your command here
+        // for example:
+        const num = interaction.options.getInteger("amount") || 1;
+        await interaction.reply("🐇".repeat(num));
+    }
+
+    register() {
+        // set the name and description shown in discord here (these are necessary)
+        // (name: string.length = 1-32, description string.length = 1-100)(both lowercase!)
```
Comment (Chicken) [inline src/commands/CommandSample.disable:20]: Description doesn't need to be lowercase?

**Review by Chicken — CHANGES_REQUESTED** [review]

File: src/commands/CommandSample.disable [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        // here you can also add things like options to your command
+        // see more under https://discord.js.org/#/docs/builders/stable/class/SlashCommandBuilder
+        return new SlashCommandBuilder()
+            .setName("bunny")
+            .setDescription("shows you a cute bunny")
+            .addIntegerOption(option => option.setName("amount").setDescription("number of bunnies").setRequired(false));
```
Comment (WilsontheWolf) [inline src/commands/CommandSample.disable:26]: Should probably set a max and min value here. min 1 max of 2000(?) would make sence

**Review by WilsontheWolf — CHANGES_REQUESTED** [review]

File: src/commands/CommandSample.disable (hunk 4 above)
Comment (StrangeGirlMurph) [reply src/commands/CommandSample.disable:20]: yup checks out
thanks

File: src/commands/CommandSample.disable (hunk 5 above)
Comment (WilsontheWolf) [reply src/commands/CommandSample.disable:26]: True.

File: src/commands/CommandSample.disable (hunk 3 above)
Comment (StrangeGirlMurph) [reply src/commands/CommandSample.disable]:
```suggestion
        super("bunny"); // the name under which the bot internally stores your command (should be the same as the named set in `register`
```

File: src/commands/CommandSample.disable (hunk 4 above)
Comment (StrangeGirlMurph) [reply src/commands/CommandSample.disable]:
```suggestion
        // (name: string.length = 1-32, description string.length = 1-100)(the name has to be lowercase!)
```

File: src/commands/CommandSample.disable [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { Command } from "../Command";
```
Comment (KommentatorForAll) [inline src/commands/CommandSample.disable:1]: CommandSample.ts.disabled

File: src/commands/CommandSample.disable [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import { CommandInteraction } from "discord.js";
+import { SlashCommandBuilder } from "@discordjs/builders";
+
+// set the class name, the export at the bottom and the file name to your desired command name (same as the one in the register function)
+class CommandName extends Command {
+    constructor() {
+        super("bunny"); // the name under which the bot internally stores your command (should be the same as the named set in `register`
```
Comment (KommentatorForAll) [inline src/commands/CommandSample.disable:8]:
```suggestion
        super("bunny"); // the name under which the bot internally stores your command (should be the same as the named set in `register`) The name must be unique.
```

**Review by Chicken — APPROVED** [review]

File: src/Command.ts [hunk 8]
```diff
@@ -8,5 +8,5 @@ export abstract class Command {
 
 	public readonly name: string;
 	abstract execute(interaction: CommandInteraction): Promise<void>;
-	abstract register(): SlashCommandBuilder;
+	abstract register(): SlashCommandBuilder | Omit<SlashCommandBuilder, "addSubcommandGroup" | "addSubcommand">;
```
Comment (KommentatorForAll) [inline src/Command.ts:11]: Why is the Omit<> thing requred?

**Review by WilsontheWolf — APPROVED** [review]

**Review by KommentatorForAll — CHANGES_REQUESTED** [review]

File: src/events/ready.ts [hunk 9]
```diff
@@ -41,7 +41,10 @@ async function updateRegisteredCommands(client: Client) {
 	}
 }
 
-function commandsEqual(c1: ApplicationCommand, c2: SlashCommandBuilder): boolean {
+function commandsEqual(
+	c1: ApplicationCommand,
+	c2: SlashCommandBuilder | Omit<SlashCommandBuilder, "addSubcommandGroup" | "addSubcommand">
```
[the remark this answers is not in the document]
Comment (WilsontheWolf) [reply src/events/ready.ts:46]: same as above

File: src/commands/PingCommand.ts [hunk 10]
```diff
@@ -12,7 +12,7 @@ class PingCommand extends Command {
 		await interaction.reply(`Pong! (Bot ping: ${botping}ms)`);
 	}
 
-	register(): SlashCommandBuilder {
+	register() {
```
[the remark this answers is not in the document]
Comment (WilsontheWolf) [reply src/commands/PingCommand.ts:15]: Actually intelisence doesn't like this one. It does like the other one though

File: src/Command.ts (hunk 8 above)
Comment (Chicken) [reply src/Command.ts:11]: Read the conversation on Discord

File: src/commands/PingCommand.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (KommentatorForAll) [reply src/commands/PingCommand.ts:15]: yep. it assumes a void or any. both things intellisens isn't able to help with. always type even if you have to use | things for it

**Review by KommentatorForAll — APPROVED** [review]
:D

**Pull request merged**